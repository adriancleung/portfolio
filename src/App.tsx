import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import Hero from './screens/Hero';
import Work from './screens/Work';
import Projects from './screens/Projects';
import firebase from './services/firebase';

const WrapperContainer = styled.div`
  background-color: #fbfcf8;
`;

const App = () => {
  const [isLoading, setIsLoading] = useState(true);

  useEffect(() => {
    firebase.config
      .fetchAndActivate()
      .then(() => {
        firebase.analytics.logEvent('config_load_success');
        setIsLoading(false);
      })
      .catch(err => firebase.analytics.logEvent('config_load_failure', err));
    firebase.analytics.logEvent('app_viewed');
  }, []);

  return isLoading ? (
    <></>
  ) : (
    <WrapperContainer>
      <Hero />
      <Work />
      <Projects />
    </WrapperContainer>
  );
};

export default App;
