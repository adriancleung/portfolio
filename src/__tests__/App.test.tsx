import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

beforeEach(() => {
  const mockIntersectionObserver = jest.fn();
  mockIntersectionObserver.mockReturnValue({
    observe: () => null,
    unobserve: () => null,
    disconnect: () => null,
  });
  window.IntersectionObserver = mockIntersectionObserver;
});

test("renders App", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
