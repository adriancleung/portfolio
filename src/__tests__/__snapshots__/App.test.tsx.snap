// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App 1`] = `
<DocumentFragment>
  <div
    class="sc-eOzmre iYeiCe"
  >
    <div
      class="sc-blHHSb jXusDg"
    >
      <div
        class="sc-gtLWhw sc-egkSDF cfhcRf dikrXs"
      >
        <div
          class="sc-fAUdSK bTPvFY"
        >
          <div
            class="sc-ivxoEo fyCtWC"
          />
          <div
            class="sc-ghWlax kUrNqE"
          >
            <div
              class="sc-kLhKbu dZrNxu"
            >
              <h1>
                Adrian Leung
              </h1>
              <p>
                 | 
              </p>
              <code>
                Software Engineer @ Intuit
              </code>
            </div>
            <div
              style="text-align: justify;"
            >
              <p>
                Thanks for dropping by!
              </p>
              <p>
                My name is 
                <strong>
                  Adrian
                </strong>
                 and I’m a 
                <strong>
                  full-stack software engineer
                </strong>
                 with an emphasis in backend development. I hold a 
                <strong>
                  BSc. - Computing Science
                </strong>
                 from the University of Alberta. I have 
                <strong>
                  2+ years of experience
                </strong>
                 working in consulting and financial technologies. My previous works include authentication, monolithic development and migration, GTM initatives, as well as AI/ML development.
              </p>
              <p>
                I’m passionate about finding innovative ways to problem-solving and using my skills to help others achieve their full potential.
              </p>
              <p>
                In my free time, I’m passionate about hiking, camping, playing volleyball, bouldering, skiing, and snowboarding, golfing, and cycling. I also enjoy hanging out with friends and exploring the great outdoors. Feel free to connect with me on my various platforms and I look forward to hearing from you!
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-dntaoT fjuryB"
        >
          <a
            class="sc-ixGGxD ghOHwq"
            href="https://linkedin.com/in/adriancleung"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-19vn4a9-MuiSvgIcon-root"
              data-testid="LinkedInIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
              />
            </svg>
          </a>
          <a
            class="sc-ixGGxD ghOHwq"
            href="https://github.com/adriancleung"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-19vn4a9-MuiSvgIcon-root"
              data-testid="GitHubIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
              />
            </svg>
          </a>
          <a
            class="sc-ixGGxD ghOHwq"
            href="https://instagram.com/adriancleung"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-19vn4a9-MuiSvgIcon-root"
              data-testid="InstagramIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.8 2h8.4C19.4 2 22 4.6 22 7.8v8.4a5.8 5.8 0 0 1-5.8 5.8H7.8C4.6 22 2 19.4 2 16.2V7.8A5.8 5.8 0 0 1 7.8 2m-.2 2A3.6 3.6 0 0 0 4 7.6v8.8C4 18.39 5.61 20 7.6 20h8.8a3.6 3.6 0 0 0 3.6-3.6V7.6C20 5.61 18.39 4 16.4 4H7.6m9.65 1.5a1.25 1.25 0 0 1 1.25 1.25A1.25 1.25 0 0 1 17.25 8 1.25 1.25 0 0 1 16 6.75a1.25 1.25 0 0 1 1.25-1.25M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3z"
              />
            </svg>
          </a>
          <a
            class="sc-ixGGxD ghOHwq"
            href="mailto:leung.c.adrian@gmail.com?subject=Inquiry%20-%20Adrian%20L"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-19vn4a9-MuiSvgIcon-root"
              data-testid="EmailOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2zm-2 0-8 5-8-5zm0 12H4V8l8 5 8-5z"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-blHHSb jXusDg"
    >
      <div
        class="sc-gtLWhw sc-khLCKb cfhcRf eFmUwZ"
      >
        <h2>
          Work Experience
        </h2>
        <div
          class="sc-dstKZu"
        >
          <div
            class="sc-keTIit jndwWP"
          >
            <h3>
              Software Engineer 2 | Intuit
            </h3>
            <span>
              August 2023 – Present
            </span>
          </div>
          <ul
            class="sc-dpBQxM kTYPVu"
          >
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Developed, trained, and deployed a machine learning model in QuickBooks Online, creating hyper-personalized dashboards with 80-90% accuracy, significantly boosting user productivity and satisfaction
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Successfully migrated from inception to deployment of APIs handling 100M+ daily requests to new capabilities, enhancing application performance and achieving multimillion-dollar savings by reducing infrastructural costs
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Led ownership of key QuickBooks Online P0 login-related assets, building end-to-end login REST APIs in microservice environments, and creating technical architecture diagrams that significantly enhanced and streamlined the authentication process
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Led initiatives to enhance observability, availability, and monitoring for critical login assets, resulting in reduced incidents and improved system reliability
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Collaborated cross-functionally to implement new authentication resources, fortifying organizational security and operational efficiency
            </li>
          </ul>
        </div>
        <div
          class="sc-dstKZu"
        >
          <div
            class="sc-keTIit jndwWP"
          >
            <h3>
              Software Engineer 1 | Intuit
            </h3>
            <span>
              May 2022 – July 2023
            </span>
          </div>
          <ul
            class="sc-dpBQxM kTYPVu"
          >
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Designed and implemented a new global login system across QuickBooks products, delivering a streamlined user experience to over 7 million customers worldwide
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Implemented rigorous testing protocols and resolved critical bugs, significantly boosting customer satisfaction and overall experience
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Engineered and integrated a login solution that simplified user experience, reducing development and infrastructure overheads while improving support efficiency
            </li>
          </ul>
        </div>
        <div
          class="sc-dstKZu"
        >
          <div
            class="sc-keTIit jndwWP"
          >
            <h3>
              Software Engineer | Medi-Scribe
            </h3>
            <span>
              August 2023 – Present
            </span>
          </div>
          <ul
            class="sc-dpBQxM kTYPVu"
          >
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Created a comprehensive documentation and drug information tool that significantly reduced the time pharmacists spend on documenting prescriptions and clinical activities
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Successfully configured and provisioned cloud infrastructures on AWS, optimizing scalability, security, and performance for Medi-Scribe, ensuring robust and reliable service delivery
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Implemented a CI/CD pipeline that streamlined builds and deployments, accelerating time-to-market for Medi-Scribe products, and enhancing team efficiency and product quality
            </li>
          </ul>
        </div>
        <div
          class="sc-dstKZu"
        >
          <div
            class="sc-keTIit jndwWP"
          >
            <h3>
              Student Software Developer | Arcurve
            </h3>
            <span>
              May 2020 – August 2021
            </span>
          </div>
          <ul
            class="sc-dpBQxM kTYPVu"
          >
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Contributed to developing, testing, and implementing critical software features for a multi-million-dollar project, ensuring functionality enhancements and on-time milestone delivery
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Successfully migrated applications to cloud infrastructures, enhancing reliability, performance, and achieving substantial operational cost savings
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Designed and implemented comparative testing platforms to ensure data reliability, streamline testing processes, and improve accuracy across applications
            </li>
            <li
              class="sc-jwIPbr kQLDuE"
            >
              Executed optimizations on existing applications, significantly improving system performance and responsiveness, thereby enhancing user experience and operational efficiency
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="sc-blHHSb jXusDg"
    >
      <div
        class="sc-gtLWhw sc-cHqXqK cfhcRf gqkZle"
      >
        <div
          class="sc-jtQUzJ joBnLE"
        >
          <h2>
            Projects
          </h2>
          <p>
            Check out some of the publicly available projects I've worked on previously!
          </p>
        </div>
        <div
          class="sc-cEzcPc jLStQj"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
